data/transactions/logic/eval.go

```diff
@@ -375,9 +375,10 @@ func NewEvalParams(txgroup []transactions.SignedTxnWithAD, proto *config.Consens
 	// Make a simpler EvalParams that is good enough to evaluate LogicSigs.
 	if apps == 0 {
 		return &EvalParams{
-			TxnGroup: txgroup,
-			Proto:    proto,
-			Specials: specials,
+			TxnGroup:  txgroup,
+			Proto:     proto,
+			Specials:  specials,
+			available: &resources{},
```

Comment:
Opening a thread to solicit feedback.

The highlighted change produces a data race due to allowing further (and unintended) execution in `RecordAD`: 
 [TestLedgerSingleTxApplyData.log](https://github.com/algorand/go-algorand/files/9789286/TestLedgerSingleTxApplyData.log).

It's unclear to me which tradeoff is preferable:
* Keep as is.  Leaves `EvalContract` susceptible to a panic when used in an unintended way.
* Defend `EvalContract` against `nil` `available`.  
  * Due to order of operations, changes the failure message from `versionCheck` to a different error about `available`.
  * We may want a more explicit flag identifying logicsig vs contract usage.
* Rework `EvalContract` to produce the same error as before while also removing `isFull`.   Unclear if we'd also want to defend against `nil` `available`.

Reply:
I don't want to make these eps "full".  That involves performing allocations that are unnecessary for the vast majority of txgroups, wich have no app calls, like a simple pay or axfer.